UCLALibrary/oral-history-staff-ui
# Sys 1230/project item export update

Some changes to import process for constancy:

- Script added for `ProjectItem` export source data in export directory
- `coverage` and `relation` added to export source data
- `coverage` and `relation` added to `import_projectitems.py`
- `import_projectitems.py` updated for tabs, sequence casting and new date mask
-  Original csv deleted, new tsv moved to export directory for consistency
- `nuke_dev_db.sh` and `reset_prod_db.sh` updated to reflect new paths
- pk integrity test added for ProjectItems, though ark should be unique as well, we do not enforce in db
- Field type updates for `create_date` and `last_modified_date`

To test:

- Start containers, run `nuke_dev_db.sh` to drop DB
- Run `docker-compose exec django python manage.py import_projectitems export_scripts/project-items-export.tsv`

To verify new date format:
`docker-compose exec django python manage.py shell`
```
(InteractiveConsole)
>>> from oh_staff_ui.models import ProjectItem
>>> p = ProjectItem.objects.get(pk=31)
>>> p.last_modified_date
datetime.datetime(2015, 9, 1, 18, 37, 51, tzinfo=datetime.timezone.utc)
```
* Your pk may be different object depending on load order

Project items should be imported without error. A warning message be presented for ark `21198/zz0008zh0f`, this is the original root/collection object and is skipped when adding items.
